src/libs/Trie/TrieNode.js

```diff
@@ -0,0 +1,50 @@
+
+class TrieNode {
+    constructor() {
+        this.children = {};
+        this.leaf = false;
+        this.metaData = {};
+    }
+
+    /**
+     * Make the current as a leaf node.
+    */
+    setCompleteWord() {
+        this.leaf = true;
+    }
+
+    /**
+    * Check if the current node is a leaf node.
+    * @returns {Boolean}
+    */
+    isCompleteWord() {
+        return this.leaf;
+    }
+
+    /**
+     * Attach additional metadata to this node.
+     *
+     * @param {Object} metaData
+     */
+    setMetaData(metaData) {
+        this.metaData = metaData;
+    }
+
+    /**
+    * Get additional metadata for this node.
+    * @return {Object} metaData
+    */
+    getMetaData() {
```

Comment:
Overall I don't think we need any of these getters or setters – this isn't Java code and we can just access or set the class members with dot notation, as you do for `children`.